packages/react-dom/src/__tests__/ReactDOMNativeEventHeuristic-test.js

```diff
@@ -281,4 +281,44 @@ describe('ReactDOMNativeEventHeuristic-test', () => {
       expect(container.textContent).toEqual('hovered');
     });
   });
+
+  // @gate experimental
+  it('should batch inside native events', async () => {
+    const root = ReactDOM.unstable_createRoot(container);
+
+    const target = React.createRef(null);
+    function Foo() {
+      const [count, setCount] = React.useState(0);
+
+      React.useEffect(() => {
+        Scheduler.unstable_yieldValue(count);
+      }, [count]);
+
+      React.useLayoutEffect(() => {
+        target.current.onclick = () => {
+          setCount(c => c + 1);
+          // Now update again, this should be batched.
+          setCount(c => c + 1);
+        };
+      });
+      return <div ref={target}>Count: {count}</div>;
+    }
+
+    await act(async () => {
+      root.render(<Foo />);
+    });
+    expect(Scheduler).toHaveYielded([0]);
+    expect(container.textContent).toEqual('Count: 0');
+
+    // Ignore act warning. We can't use act because it forces batched updates.
+    spyOnDev(console, 'error');
```

Comment:
I don't love this but I cannot get toErrorDev to work.

Reply:
Make sure you assert that only the expected error is logged, though

```js
expect(console.error).toHaveBeenCalledTimes(1);
expect(console.error.calls.argsFor(0)[0]).toContain(hookWarningMessage);
```